microsoft/testsuites/core/dns.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from __future__ import annotations
+
+from retry import retry
+
+from lisa import (
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    UnsupportedDistroException,
+    simple_requirement,
+)
+from lisa.operating_system import Debian, Posix
+from lisa.tools import Ping
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    This test suite covers DNS name resolution functionality.
+    """,
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class Dns(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case check DNS name resolution by ping bing.com.
+        """,
+        priority=1,
+    )
+    def verify_dns_name_resolution(self, node: Node) -> None:
```

Comment:
The test case behavior and name are inconsistent. 
1. The DNS can be tested without any other action, it's worth to have a case. 
2. The unattented-upgrade can be another test case. After upgraded, we can test the dns, reboot and other P0 scenarios.